opentelemetry-sdk/src/metrics/mod.rs

```diff
@@ -2412,6 +2414,123 @@ mod tests {
             empty_attrs_data_point.value, 6,
             "Empty attributes value should be 3+3=6"
         );
+
+        // Phase 2 - for delta temporality, after each collect, data points are cleared
+        if temporality == Temporality::Delta {
+            test_context.reset_metrics();
+            // This should be aggregated normally, and not go into overflow.
+            counter.add(100, &[KeyValue::new("A", "foo")]);
+            counter.add(100, &[KeyValue::new("A", "another")]);
+            counter.add(100, &[KeyValue::new("A", "yet_another")]);
+            test_context.flush_metrics();
+
+            let MetricData::Sum(sum) = test_context.get_aggregation::<u64>("my_counter", None)
+            else {
+                unreachable!()
+            };
+            assert_eq!(sum.data_points.len(), 3);
+
+            let data_point = find_sum_datapoint_with_key_value(&sum.data_points, "A", "foo")
+                .expect("point expected");
+            assert_eq!(data_point.value, 100);
+
+            let data_point = find_sum_datapoint_with_key_value(&sum.data_points, "A", "another")
+                .expect("point expected");
+            assert_eq!(data_point.value, 100);
+
+            let data_point =
+                find_sum_datapoint_with_key_value(&sum.data_points, "A", "yet_another")
+                    .expect("point expected");
+            assert_eq!(data_point.value, 100);
+        }
+    }
+
+    fn counter_aggregation_overflow_helper_custom_limit(temporality: Temporality) {
+        // Arrange
+        let cardinality_limit = Arc::new(2300);
```

Comment:
There is no need to use `Arc` and related cloning, dereferencing etc. here. A simple `usize` typed variable would work.

```suggestion
        let cardinality_limit: usize = 2300;
```